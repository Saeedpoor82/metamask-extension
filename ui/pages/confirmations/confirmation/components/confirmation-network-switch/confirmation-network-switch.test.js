import React from 'react';
import { useSelector } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { getTheme } from '../../../../../selectors';
import mockState from '../../../../../../test/data/mock-state.json';
import { renderWithProvider } from '../../../../../../test/lib/render-helpers';
import ConfirmationNetworkSwitch from './confirmation-network-switch';

// Mock Redux store
jest.mock('react-redux', () => {
  const actual = jest.requireActual('react-redux');

  return {
    ...actual,
    useSelector: jest.fn(),
  };
});

const mockPalmNetwork = {
  ...mockState,
  metamask: {
    ...mockState.metamask,
    providerConfig: {
      chainId: '0x7',
    },
    networkConfigurations: {
      ...mockState.metamask.networkConfigurations,
      testNetworkConfigurationId: {
        rpcUrl: 'https://testrpc.com',
        chainId: '0x7',
        nickname: 'Custom Mainnet RPC',
        type: 'rpc',
        id: 'testNetworkConfigurationId',
        rpcPrefs: {
          imageUrl: './some_image',
        },
      },
    },
  },
  confirm: {
    currentConfirmation: { id: '1', msgParams: {} },
  },
};

// Mock networks
const mockFromNetwork = {
  chainId: '1',
  nickname: 'Ethereum',
};
const mockToNetwork = {
  chainId: '137',
  nickname: 'Polygon',
};

const generateUseSelectorRouter = (opts) => (selector) => {
  if (selector === getTheme) {
    return opts;
  }
  return undefined;
};

const mockStore = configureMockStore([])(mockPalmNetwork);

describe('ConfirmationNetworkSwitch', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProvider(
      <ConfirmationNetworkSwitch
        toNetwork={mockToNetwork}
        fromNetwork={mockFromNetwork}
      />,
      mockStore,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders network switch component with given networks', () => {
    useSelector.mockImplementation(generateUseSelectorRouter('dark'));
    const { getByTestId } = renderWithProvider(
      <ConfirmationNetworkSwitch
        toNetwork={mockToNetwork}
        fromNetwork={mockFromNetwork}
      />,
      mockStore,
    );

    expect(getByTestId('network-switch-from-network')).toHaveTextContent(
      'Ethereum',
    );
    expect(getByTestId('network-switch-to-network')).toHaveTextContent(
      'Polygon',
    );
  });
});
