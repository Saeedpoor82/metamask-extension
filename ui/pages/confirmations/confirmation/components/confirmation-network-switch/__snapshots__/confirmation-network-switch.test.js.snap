// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationNetworkSwitch should match snapshot 1`] = `
<div>
  <div
    class="mm-box confirmation-network-switch mm-box--margin-top-8 mm-box--display-flex mm-box--justify-content-center mm-box--height-full"
  >
    <div
      class="mm-box confirmation-network-switch__icon mm-box--display-block"
    >
      <div
        class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xl mm-avatar-network mm-text--body-lg-medium mm-text--text-transform-uppercase mm-box--margin-bottom-2 mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
      >
        E
      </div>
      <p
        class="mm-box mm-text mm-text--body-md mm-box--display-flex mm-box--justify-content-center mm-box--color-text-default"
        data-testid="network-switch-from-network"
      >
        Ethereum
      </p>
    </div>
    <div
      class="mm-box confirmation-network-switch__center-icon mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center"
    >
      <i
        class="fa fa-angle-right fa-lg confirmation-network-switch__check"
      />
      <div
        class="confirmation-network-switch__dashed-line"
      />
    </div>
    <div
      class="mm-box confirmation-network-switch__icon mm-box--display-block"
    >
      <div
        class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xl mm-avatar-network mm-text--body-lg-medium mm-text--text-transform-uppercase mm-box--margin-bottom-2 mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
      >
        P
      </div>
      <p
        class="mm-box mm-text mm-text--body-md mm-box--display-flex mm-box--justify-content-center mm-box--color-text-default"
        data-testid="network-switch-to-network"
      >
        Polygon
      </p>
    </div>
  </div>
</div>
`;
